query/query.go

```diff
@@ -149,20 +149,17 @@ type Results interface {
 	Next() <-chan Result      // returns a channel to wait for the next result
 	NextSync() (Result, bool) // blocks and waits to return the next result, second parameter returns false when results are exhausted
 	Rest() ([]Entry, error)   // waits till processing finishes, returns all entries at once.
-	Close() error             // client may call Close to signal early exit
-
-	// Process returns a goprocess.Process associated with these results.
-	// most users will not need this function (Close is all they want),
-	// but it's here in case you want to connect the results to other
-	// goprocess-friendly things.
-	Process() goprocess.Process
+	Close()                   // client may call Close to signal early exit
+	Done() <-chan struct{}    // signals that Results is closed
```

Comment:
```suggestion
	Done() <-chan struct{}    // signals that the Result channel returned by Next() has been closed due to early termination (call to Close).
```

Is this the actual meaning?

Also, `Next()` returns a channel to read the results from. Current comments sounds as if you need to call Next() separately for every result and read it from the returned channel every time.

Reply:
It means that any goroutine started bu `ResultsWithContext` or `ResultsWIthChan` has exited (no more writing to the results channel) and the results channel is closed. It allows waiting on `Results` to complete, without having to read `Next`.